src/auth/jwt.refreshtoken.strategy.ts

```diff
@@ -0,0 +1,27 @@
+import { PassportStrategy } from '@nestjs/passport';
+import { Strategy, ExtractJwt } from 'passport-jwt';
+import { Injectable, UnauthorizedException, Body } from '@nestjs/common';
+import { UsersService } from '../users/users.service';
+import { jwtConstants } from './constants';
+ 
+@Injectable()
+export class JwtRefreshTokenStrategy extends PassportStrategy(Strategy, 'jwt-refreshtoken') {
+    constructor() {
+        super({
+            jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+            ignoreExpiration: false,
+            secretOrKey: jwtConstants.refresh_secret,
+        });
+    }
+  //   async validate(username: string, password:string): Promise<any> {
+  //     const user = await this.authService.validateUser(username, password);
+  //     if (!user) {
+  //         throw new UnauthorizedException();
+  //     }
+  //     return user;
+  // }
+    async validate(payload: any) {
+      return { employeeId: payload.sub.employeeId, role: payload.sub.role, companyId: payload.sub.companyId, email: payload.username };
```

Comment:
More of a question: This should be returning user information? Wouldn't this make it act as a regular JWT token? I thought the purpose of the refresh token was to serve a new JWT token?

I'm not very knowledgeable on refresh tokens, so I'm probably missing some key info that makes this make sense.

Reply:
I and Sahil had a similar conversation. I initially stored refresh token in DB but we don't want to make DB schema any bigger. Thus, we decided to make refresh token exactly the same as access token. However, we only pass refresh token only when access token expires. Thus, there is less chance that hackers intercept refresh token. Hope this help.